src/atom_finder/classifier/curly-braces.clj

```diff
@@ -0,0 +1,30 @@
+(in-ns 'atom-finder.classifier)
+(import '(org.eclipse.cdt.core.dom.ast IASTCompoundStatement IASTIfStatement IASTForStatement IASTWhileStatement))
+
+
+(defn curly-braces?
+  "Does this note represent curly brace?"
+  [node]
+  (instance? IASTCompoundStatement node))
+  
+
+(defn curly-braciable-statement?
+  "Does this AST node an statement that should be followed by curly braces?"
+  [node]
+  (let [statement #{;statement that should be followed by compound statements (curly braces)
+                    IASTIfStatement IASTForStatement IASTWhileStatement}]
+      (cond
+        (leaf? node) false
+        (some #(instance? % node) statement) (not-any? true? (map curly-braces? (children node)))
+        :else false)))
+          
+(def curly-braces-atom? curly-braciable-statement?)
+                                               
+
+(defn curly-braces-atoms
+  "Return all AST nodes that are missing curly braces"
+  [node]
+  (cond
+    (leaf? node) nil
+    (curly-braciable-statement? node) [node]
+    :else (mapcat curly-braces-atoms (children node))))
```

Comment:
default-finder is mostly just an alias to atoms-in-tree which you can read here:
https://github.com/dgopstein/atom-finder/blob/master/src/atom_finder/util.clj#L323
basically it just does a post-order traversal (https://en.wikipedia.org/wiki/Tree_traversal#Post-order_2) of the AST applying the classifier to every node.

I'm not sure there's an advantage to the one I wrote for logic-as-control-flow, it might've just been there before I'd made default-finder. If you're curious though, it'd be interesting to benchmark the two and see which is faster!